messaging/src/main/java/org/springframework/security/messaging/util/matcher/MessageMatcherEntry.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.messaging.util.matcher;
+
+/**
+ * A rich object for associating a {@link MessageMatcher} to another object.
+ *
+ * @author Josh Cummings
+ * @since 5.7
+ */
+public class MessageMatcherEntry<T> {
```

Comment:
What are the benefits of this being a top level class vs inner class of MessageMatcherDelegatingAuthorizationManager

Reply:
Inlined.